apps/token-e2e/src/integration/wallet-eth.cy.js

```diff
@@ -0,0 +1,241 @@
+      });
+
+      it('should have locked currency visible', function () {
+        cy.get(ethWallet.vegaInVesting).within(() => {
+          cy.get(ethWallet.currencyLocked)
+            .should('be.visible')
+            .invoke('text')
+            .should('match', /\d{0,3}(,\d{3})*\.\d{2}$/);
+        });
+      });
+
+      it('should have unlocked currency visible', function () {
+        cy.get(ethWallet.vegaInVesting).within(() => {
+          cy.get(ethWallet.currencyUnlocked)
+            .should('be.visible')
+            .invoke('text')
+            .should('match', /\d{0,3}(,\d{3})*\.\d{2}$/);
+        });
+      });
+
+      it('should match total & locked/unlocked currency value', function () {
+        cy.get(ethWallet.vegaInVesting)
+          .within(() => {
+            [
+              [ethWallet.currencyValue, 'currencyValue'],
+              [ethWallet.currencyLocked, 'currencyLocked'],
+              [ethWallet.currencyUnlocked, 'currencyUnlocked'],
+            ].forEach(([element, alias]) => {
+              cy.get(element).invoke('text').as(alias);
+            });
+          })
+          .then(function () {
+            const [value, locked, unlocked] = [
+              this.currencyValue,
+              this.currencyLocked,
+              this.currencyUnlocked,
+            ].map((v) => parseFloat(v.replace(/,/g, '')));
+            expect(value).to.equal(locked + unlocked);
+          });
+      });
+    });
+
+    describe('VEGA IN WALLET', function () {
+      it('should have currency title visible', function () {
+        cy.get(ethWallet.vegaInWallet).within(() => {
+          cy.get(ethWallet.currencyTitle)
+            .should('be.visible')
+            .and('have.text', 'VEGAIn Wallet');
+        });
+      });
+
+      it('should have currency value visible', function () {
+        cy.get(ethWallet.vegaInWallet).within(() => {
+          cy.get(ethWallet.currencyValue)
+            .should('be.visible')
+            .invoke('text')
+            .should('match', /\d{0,3}(,\d{3})*\.\d{18}$/);
+        });
+      });
+
+      it('should have progress bar visible', function () {
+        cy.get(ethWallet.vegaInWallet).within(() => {
+          cy.get(ethWallet.progressBar).should('be.visible');
+        });
+      });
+
+      it('should have locked currency visible', function () {
+        cy.get(ethWallet.vegaInWallet).within(() => {
+          cy.get(ethWallet.currencyLocked)
+            .should('be.visible')
+            .invoke('text')
+            .should('match', /\d{0,3}(,\d{3})*\.\d{2}$/);
+        });
+      });
+
+      it('should have unlocked currency visible', function () {
+        cy.get(ethWallet.vegaInWallet).within(() => {
+          cy.get(ethWallet.currencyUnlocked)
+            .should('be.visible')
+            .invoke('text')
+            .should('match', /\d{0,3}(,\d{3})*\.\d{2}$/);
+        });
+      });
+
+      it('should match total & locked/unlocked currency value', function () {
```

Comment:
or since we know our data upfront why not get it up front - this gives the test file a much cleaner look?

```
context('Ethereum Wallet - verify elements on widget', function () {
  before('visit token home page', function () {
    cy.visit('/');
    cy.get(ethWallet.walletContainer, { timeout: 10000 }).should('be.visible');
    cy.get(ethWallet.vegaInWallet).within(() => {
      cy.get(ethWallet.currencyValue).invoke('text').convertTokenValueToInt().as(currencyValue)
      cy.get(ethWallet.currencyLocked).invoke('text').convertTokenValueToInt().as(currencyLocked)
      cy.get(ethWallet.currencyUnlocked).invoke('text').convertTokenValueToInt().as(currencyUnlocked)
    })
  });
  ```
  ```
  it('should match total & locked/unlocked currency value2', function () {
        expect(this.currencyValue).to.equal(this.currencyLocked + this.currencyUnlocked);
  });
  ```
  ```
  Cypress.Commands.add('convertTokenValueToInt', {prevSubject: true}, (subject) => {
    return parseFloat(subject.replace(/,/g, ''))
  })

Reply:
It's a matter of preference I think :D I don't like this 'then' chaining, it reminds me callback hell.
Also, as I do mostly frontend validations here and I don't really mind what's set up in capsule, so I would rather stick with the solution I made (then I don't need to care if all the data is set up correctly, and the validations will still pass). 

I like the idea of creating convert function, though!